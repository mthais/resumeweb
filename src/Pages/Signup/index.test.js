import React from 'react'
import { MemoryRouter } from 'react-router'
import { mount } from 'enzyme'
import { SignupPage } from '.'

describe('SignupPage', () => {
  let wrapper
  let props

  beforeEach(() => {
    props = {
      isFetchingRegister: false,
      handleRegister: jest.fn()
    }

    wrapper = mount(
      <MemoryRouter initialEntries={ [ '/' ] }>
        <SignupPage { ...props } />
      </MemoryRouter>
    )
  })

  it('should match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('when submit is called', () => {
    const data = { name: 'user', email: 'user@test.com', password: '123' }

    beforeEach(async () => {
      await wrapper.find(SignupPage).instance().submit(data)
    })

    it('should calls handleSubmit function with data', () => {
      expect(props.handleRegister).toBeCalledWith(data)
    })
  })
})
