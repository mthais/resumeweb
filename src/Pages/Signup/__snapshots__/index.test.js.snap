// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage should match with snapshot 1`] = `"<div class=\\"col\\"><h2 class=\\"text-center page-title\\">Bem Vindo!</h2><div class=\\"row\\"><div class=\\"col\\"><button class=\\"py-0 pt-1 btn btn-primary btn-block\\"><div class=\\"no-gutters row\\"><div class=\\"text-right col\\"><ion-icon style=\\"font-size: 28px;\\" name=\\"logo-linkedin\\"></ion-icon></div><div class=\\"text-left pt-1 col\\">Linked in</div></div></button></div></div><div class=\\"divider\\"><span>OU</span></div><form><div class=\\"position-relative form-group\\"><label for=\\"name\\" class=\\"mb-0\\">Nome</label><div><input name=\\"name\\" required=\\"\\" maxlength=\\"20\\" id=\\"name\\" type=\\"text\\" class=\\"form-control form-control\\" value=\\"\\"><div class=\\"invalid-feedback\\"></div></div></div><div class=\\"position-relative form-group\\"><label for=\\"email\\" class=\\"mb-0\\">Email</label><div><input name=\\"email\\" id=\\"email\\" placeholder=\\"email@exemplo.com\\" type=\\"email\\" class=\\"form-control form-control\\" value=\\"\\"><div class=\\"invalid-feedback\\"></div></div></div><div class=\\"position-relative form-group\\"><label for=\\"password\\" class=\\"mb-0\\">Senha</label><div><input name=\\"password\\" id=\\"password\\" type=\\"password\\" class=\\"form-control form-control\\" value=\\"\\"><div class=\\"invalid-feedback\\"></div></div></div><div class=\\"position-relative form-check\\"><label class=\\"text-wrapper form-check-label\\"><div><input required=\\"\\" name=\\"policies\\" id=\\"policies\\" type=\\"checkbox\\" class=\\"form-check-input\\" value=\\"S\\"><div class=\\"invalid-feedback\\"></div></div>Para cadastrar, você concorda com nossa<a class=\\"link\\" href=\\"/\\"> Politica de privacidade</a> e<a class=\\"link\\" href=\\"/\\"> Termos de uso</a></label></div><div class=\\"pt-3 row\\"><div class=\\"col\\"><a class=\\"link\\" href=\\"/login\\">Eu já tenho uma conta!</a></div><div class=\\"text-right col\\"><button type=\\"submit\\" disabled=\\"\\" class=\\"px-4 btn btn-primary disabled\\">Cadastrar</button></div></div></form></div>"`;
